### PR TITLE
dropping @ValidationException feature

### PR DESCRIPTION
**Pull request by Turini**
There is no docs about it and IMHO we can drop this hidden feature.
What do you think? It can be useful for you? If merged it closes https://github.com/caelum/vraptor4/issues/949.

### CODE REVIEW

**Comment by rponte** [comment]
If I'm not wrong, this annotation is used to register validation error messages instead of showing Tomcat 500 error page when business code throws an exception (annotated with `@ ValidationException `) from controller.

**Comment by renanigt** [comment]
If drop it, it's necessary change the docs about `Testing Components and Controllers`, since `ValidationException.class` is expected in some tests like this:

```
@Test(expected = ValidationException.class)
public void shouldThrowValidationException() {
    controller.add(new Person());
}
```

**Comment by Turini** [comment]
@rponte you're right, but the point is: is it used/useful? 

@renanigt I'm dropping the annotation and not the class.

**Comment by renanigt** [comment]
Oh, true @Turini... Sorry.

**Comment by rponte** [comment]
I like the idea because we can mix it with `validator.onError*()` methods
without the need to write try-catchs in every controller's actions. Does it
make sense?

But you're right, we need a better documentation about this feature!

On Wed, Feb 25, 2015 at 4:02 PM, Renan Montenegro <notifications@github.com>
wrote:

> Oh, true @Turini <https://github.com/Turini>... Sorry.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/caelum/vraptor4/pull/951#issuecomment-76031360>.
>



-- 
Rafael Ponte
TriadWorks | Formação Java
http://cursos.triadworks.com.br

File: vraptor-core/src/test/java/br/com/caelum/vraptor/http/iogi/IogiParametersProviderTest.java [hunk 1]
```diff
@@ -745,11 +735,11 @@ public void setStreet(String street) {
 
 	public static class AngryCat {
```
Comment (lucascs) [inline vraptor-core/src/test/java/br/com/caelum/vraptor/http/iogi/IogiParametersProviderTest.java]: do we need this class?

**Comment by lucascs** [comment]
:ok_hand:, for a major release.

File: vraptor-core/src/test/java/br/com/caelum/vraptor/http/iogi/IogiParametersProviderTest.java (hunk 1 above)
Comment (Turini) [inline vraptor-core/src/test/java/br/com/caelum/vraptor/http/iogi/IogiParametersProviderTest.java]: removed here c17a92e8ee65c9d1f9ddc79cf776788081bfe807, thanks!

**Comment by Turini** [comment]
> [...] mix it with `validator.onError*()` methods without the need 
to write try-catchs in every controller's actions. Does it make sense?

Yes, it does! I'm just not convinced that someone will use it anyway :)

I'll wait more reviews before delete or add docs. thoughts `@everyone`?

**Comment by nykolaslima** [comment]
I never used this feature before but for me it doesn't makes sense.

If users want to deal with some specific exception they can do it with an interceptor or something like that.

**Comment by Turini** [comment]
I think we can drop it and restore when (and if) someone ask for this feature.
If it don't get any downvote until monday, I'll merge here and ship it with `4.2`.

**Comment by rponte** [comment]
Go ahead, Turini!

In the worst case we can create a plugin for this! :-)

On Fri, Mar 6, 2015 at 8:54 AM, Rodrigo Turini <notifications@github.com>
wrote:

> I think we can drop it and restore when (and if) someone ask for this
> feature.
> If it don't get any downvote until monday, I'll merge here and ship it
> with 4.2.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/caelum/vraptor4/pull/951#issuecomment-77548099>.
>



-- 
Rafael Ponte
TriadWorks | Formação Java
http://cursos.triadworks.com.br

**Comment by nykolaslima** [comment]
:+1:

**Comment by Turini** [comment]
Nice, thank you guys!

**Pull request merged**